src/api/crypto/blockchain/Imp_blockchain.cpp

```diff
@@ -381,72 +409,82 @@ auto BlockchainImp::ProcessMergedContact(
     return true;
 }
 
-auto BlockchainImp::ProcessTransaction(
+auto BlockchainImp::ProcessTransactions(
     const opentxs::blockchain::Type chain,
-    const opentxs::blockchain::block::bitcoin::Transaction& in,
+    Set<std::shared_ptr<opentxs::blockchain::bitcoin::block::Transaction>>&& in,
     const PasswordPrompt& reason) const noexcept -> bool
 {
     const auto& db = api_.Network().Blockchain().Internal().Database();
-    auto copy = in.clone();
+    const auto& log = LogTrace();
+    auto lock = Lock{lock_};
 
-    OT_ASSERT(copy);
+    for (auto& pTX : in) {
+        OT_ASSERT(pTX);
```

Comment:
assert usually is expensive, I hope this loop always will be short

Reply:
MT-81

Reply:
MT-79 refers to a line number, which may change. Add a TODO note, please.

Reply:
classname added